documentation/src/main/asciidoc/public/reference/_search-dsl-aggregation.adoc

```diff
@@ -324,6 +324,117 @@ See <<search-dsl-argument-type>> for more information.
+that is, the aggregation will be applied to the entire hit set.
+====
+
+[IMPORTANT]
+====
+The fields that are targeted by the aggregation function must be declared <<mapping-directfieldmapping-aggregable,aggregable>>.
+====
+
+=== Sum metric aggregation
+
+The `sum` aggregation sums up the field values.
+
+.Sum the prices all of the science fiction books
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/aggregation/AggregationDslIT.java[tags=sums]
+----
+<1> Define the target field path to which you want to apply the aggregation function and the expected returned type.
+====
+
+[NOTE]
+====
+You can always use the field type for the expected returned type.
+Another option is to set `Double.class` for the expected returned type and you can always do it
+even if the field is not double typed.
+====
+
+=== Min metric aggregation
+
+The `min` aggregation provides the minimum value among the field values.
+
+.Find the min release date among all the science fiction books
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/aggregation/AggregationDslIT.java[tags=min]
+----
+<1> Define the target field path to which you want to apply the aggregation function and the expected returned type.
+====
+
+=== Max metric aggregation
+
+The `max` aggregation provides the maximum value among the field values.
+
+.Find the max release date among all the science fiction books
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/aggregation/AggregationDslIT.java[tags=max]
+----
+<1> Define the target field path to which you want to apply the aggregation function and the expected returned type.
+====
+
+=== Count metric aggregation
+
+The `count` aggregation counts the field values.
+
+.Count anytime the price field has a value among all the science fiction books
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/aggregation/AggregationDslIT.java[tags=count]
+----
+<1> Define the target field path to which you want to apply the aggregation function. For this function a `Long.class` value is always returned.
+====
+
+=== Count distinct metric aggregation
+
+The `count distinct` aggregation counts the different field values.
+
+.Count anytime the price field has a different value among all the science fiction books
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/aggregation/AggregationDslIT.java[tags=count-distinct]
+----
+<1> Define the target field path to which you want to apply the aggregation function. For this function a `Long.class` value is always returned.
+====
+
+=== Avg metric aggregation
+
+Applies the SQL `avg` aggregation function on a given numeric or temporal field.
```

Comment:
I think we've missed this one as we were replacing the sql-based definition in the previous iteration 🙈. Maybe
```suggestion
The `avg` aggregation calculates the average value of a given numeric or temporal field among the selected documents.
```